aspnetcore/security/enforcing-ssl.md

```diff
@@ -46,26 +47,55 @@ The preceding highlighted code:
 
-We recommend using temporary redirects rather than permanent redirects, as link caching can cause unstable behavior in development environments. We recommend using [HSTS](#hsts) to signal to clients that only secure resource requests should be sent to the app (only in production).
+We recommend using temporary redirects rather than permanent redirects. Link caching can cause unstable behavior in development environments. If you prefer to send a permanent redirect status code when the app is in a non-Development environment, see the [Configure permanent redirects in production](#configure-permanent-redirects-in-production) section. We recommend using [HSTS](#http-strict-transport-security-protocol-hsts) to signal to clients that only secure resource requests should be sent to the app (only in production).
 
-> [!WARNING]
-> A port must be available for the middleware to redirect to HTTPS. If no port is available, redirection to HTTPS doesn't occur. The HTTPS port can be specified using any of the following approaches:
->
-> * Set `HttpsRedirectionOptions.HttpsPort`.
-> * Set the `ASPNETCORE_HTTPS_PORT` environment variable.
-> * In development, set an HTTPS URL in *launchsettings.json*.
-> * Configure an HTTPS URL endpoint for [Kestrel](xref:fundamentals/servers/kestrel) or [HTTP.sys](xref:fundamentals/servers/httpsys).
->
-> When Kestrel or HTTP.sys is used as a public-facing edge server, Kestrel or HTTP.sys must be configured to listen on both:
->
-> * The secure port where the client is redirected (typically, 443 in production and 5001 in development).
-> * The insecure port (typically, 80 in production and 5000 in development).
->
-> The insecure port must be accessible by the client in order for the app to receive an insecure request and redirect it to the secure port.
->
-> Any firewall between the client and server must also have the ports open for traffic.
->
-> For more information, see [Kestrel endpoint configuration](xref:fundamentals/servers/kestrel#endpoint-configuration) or <xref:fundamentals/servers/httpsys>.
+### Port configuration
+
+A port must be available for the middleware to redirect an insecure request to HTTPS. If no port is available:
+
+* Redirection to HTTPS doesn't occur.
+* The middleware logs the warning "Failed to determine the https port for redirect."
+
+Specify the HTTPS port using any of the following approaches:
+
+* Set [HttpsRedirectionOptions.HttpsPort](#options).
+* Set the `ASPNETCORE_HTTPS_PORT` environment variable or [https_port Web Host configuration setting](xref:fundamentals/host/web-host#https-port):
+
+  **Key**: https_port  
+  **Type**: *string*  
+  **Default**: A default value isn't set.  
+  **Set using**: `UseSetting`  
+  **Environment variable**: `<PREFIX_>HTTPS_PORT` (The prefix is `ASPNETCORE_` when using the [Web Host](xref:fundamentals/host/web-host).)
+
+  ```csharp
```

Comment:
Thoughts on indicating that this code belongs in *Program.cs*?

Reply:
> Thoughts on indicating that this code belongs in Program.cs?

I agree ... there's very little cost to helping new devs understand where things go.